text/0000-regex-1.0.md

```diff
@@ -0,0 +1,980 @@
+
+As of now, `regex` has several dependencies:
+
+* `aho-corasick`
+* `memchr`
+* `thread_local`
+* `regex-syntax`
+* `utf8-ranges`
+
+All of them except for `thread_local` were written by this author, and were
+primarily motivated for use in the `regex` crate. They were split out because
+they seem generally useful.
+
+There may be other things in `regex` (today or in the future) that may also be
+helpful to others outside the strict context of `regex`. Is it beneficial to
+split such things out and create a longer list of dependencies? Or should we
+keep `regex` as tight as possible?
+
+## Exposing more internals
+[exposing-more-internals]: #exposing-more-internals
+
+It is conceivable that others might find interest in the regex compiler or more
+lower level access to the matching engines. We could do something similar to
+`regex-syntax` and expose some internals in a separate crate. However, there
+isn't a pressing desire to do this at the moment, and would probably require a
+good deal of work.
+
+# Breaking changes
+[breaking-changes]: #breaking-changes
+
+This section of the RFC lists all breaking changes between `regex 0.1` and the
+API proposed in this RFC.
+
+* `bytes::Regex` enables the Unicode flag by default. Previously, it disabled
+  it by default. The flag can be disabled in the pattern with `(?-u)`.
+* The definition of the `Replacer` trait was completely re-worked. Namely, its
+  API inverts control of allocation so that the caller must provide a `String`
+  to write to. Previous implementors will need to examine the new API. Moving
+  to the new API should be straight-forward.
+* The `is_empty` method on `Captures` was removed since it always returns
+  `false` (because every `Captures` has at least one capture group
+  corresponding to the entire match).
+* The `PartialEq` and `Eq` impls on `Regex` were removed. If you need this
+  functionality, add a newtype around `Regex` and write the corresponding
+  `PartialEq` and `Eq` impls.
+* The lifetime parameters for the `iter` and `iter_named` methods on
+  `Captures` were fixed. The corresponding iterator types, `SubCaptures` and
+  `SubCapturesNamed`, grew an additional lifetime parameter.
+* The constructor, `Regex::with_size_limit`, was removed. It can be replaced
+  with use of `RegexBuilder`.
+* The `is_match` free function was removed. Instead, compile a `Regex`
+  explicitly and call the `is_match` method.
+* Many iterator types were renamed. (e.g., `RegexSplits` to `SplitsIter`.)
+* Replacements now return a `Cow<str>` instead of a `String`. Namely, the
+  subject text doesn't need to be copied if there are no replacements. Callers
+  may need to add `into_owned()` calls to convert the `Cow<str>` to a proper
+  `String`.
+* The `Error` type no longer has the `InvalidSet` variant, since the error is
+  no longer possible. Its `Syntax` variant was also modified to wrap a `String`
+  instead of a `regex_syntax::Error`. If you need access to specific parse
```

Comment:
Maybe it could wrap a  type defined in `regex` that internally contains a `regex_syntax::Error` (the type may just implement `Error`, `Display` etc. for now), so we can choose to expose more detailed info in future if we want.